gobblin-core/src/main/java/gobblin/source/extractor/partition/Partitioner.java

```diff
@@ -119,7 +119,8 @@ private long getLowWatermark(ExtractType extractType, WatermarkType watermarkTyp
       int deltaForNextWatermark) {
     long lowWatermark = ConfigurationKeys.DEFAULT_WATERMARK_VALUE;
     if (this.isFullDump() || this.isWatermarkOverride()) {
-      lowWatermark = Utils.getLongWithCurrentDate(this.state.getProp(ConfigurationKeys.SOURCE_QUERYBASED_START_VALUE));
+      lowWatermark = Utils.getLongWithCurrentDate(this.state.getProp(ConfigurationKeys.SOURCE_QUERYBASED_START_VALUE),
+          this.state.getProp(ConfigurationKeys.SOURCE_TIMEZONE));
```

Comment:
Thanks for your feedback! I didn't pass the default value because Utils#getCurrentTime already handles it. I didn't touch line 145, but can change it accordingly.